opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -356,12 +356,34 @@ def __init__(
     ):
         self.instrumentation_info = instrumentation_info
         self.processor = Processor(source.stateful, source.resource)
-        self.metrics = set()
-        self.observers = set()
+        self.metrics = {}
+        self.observers = {}
         self.metrics_lock = threading.Lock()
         self.observers_lock = threading.Lock()
         self.view_manager = ViewManager()
 
+    def _register_metric(self, instrument: metrics_api.Metric):
+        name = instrument.name.strip().lower()
+        with self.metrics_lock:
+            if name in self.metrics:
```

Comment:
Doesn't this now mean that I can have a `metric` named "test" and an `observer` named "test?

Reply:
Yes. As discussed in another thread I will make change to have one container for both metrics and observers. This also breaks few things in contrib repo. I will be pushing all these changes once.